Report/score.R

```diff
@@ -58,12 +43,24 @@ save_score_cards <- function(score_card, geo_type = c("state", "nation"),
       filter(geo_value == "us")
   }
 
+  output_file_name <- generate_score_card_file_path(geo_type, signal_name, output_dir)
   saveRDS(score_card,
     file = output_file_name,
     compress = "xz"
   )
 }
 
+save_score_cards_wrapper <- function(score_card, geo_type, signal_name, output_dir) {
+  if (signal_name %in% unique(score_card["signal"])) {
+    print(paste("Saving", geo_type, type_map[[signal_name]], "..."))
+    save_score_cards(score_card, geo_type,
+      signal_name = signal_name, output_dir = output_dir
+    )
+  } else {
+    stop(paste("No", signal_name, "available at the", geo_type, "level"))
```

Comment:
is halting here the intended behavior, or should this be a warning that gets collected after all the available signal+geo combos have been processed?

Reply:
I think this is serious enough to stop for. I changed this wrapper function to return a status (0 or message if something went wrong). The status for all such calls are then bundled and raised as an error at the same time so we get more insight into which signal/geos were missing.